scipy/_build_utils/_generate_blas_wrapper.py

```diff
@@ -61,31 +61,29 @@ def generate_decl_wrapper(name, return_type, argnames, argtypes, accelerate):
     blas_macro, blas_name = get_blas_macro_and_name(name, accelerate)
     return f"""
 {c_return_type} {blas_macro}({blas_name})({param_list});
-{c_return_type} F_FUNC({name},{name.upper()})({param_list}){{
+HIDDEN_SYMBOL {c_return_type} F_FUNC({name},{name.upper()})({param_list}){{
     return {blas_macro}({blas_name})({argnames});
 }}
 """
 
 
-def generate_file_wrapper(sigs, lib_name, accelerate, outdir):
+def generate_file_wrapper(sigs, accelerate):
     """
-    Returns a dictionary of wrapper file paths to their string contents.
-    Each BLAS/LAPACK function gets a separate wrapper file. All wrapper files
-    are compiled into an archive that allows the linker to selectively include
-    only the necessary symbols for each binary.
+    Returns text of file containing wrappers for all BLAS/LAPACK functions.
     """
-    if lib_name == 'BLAS':
-        preamble = [C_COMMENT, C_PREAMBLE, CPP_GUARD_BEGIN]
-    elif lib_name == 'LAPACK':
-        preamble = [C_COMMENT, C_PREAMBLE, LAPACK_DECLS, CPP_GUARD_BEGIN]
-    wrappers = {}
+    file_text = [C_COMMENT, C_PREAMBLE, LAPACK_DECLS, CPP_GUARD_BEGIN]
+    # Let linker strip symbols from final binaries
+    file_text.append("""
+#if __GNUC__ >= 4
+#define HIDDEN_SYMBOL  __attribute__ ((visibility ("hidden")))
```

Comment:
From what I can tell, we could put this in `npy_cblas.h`, which is included directly in the other two files, and gets picked up here through:
https://github.com/scipy/scipy/blob/438e88f5e1752f49b5c760e15c94c48f26d83e4d/scipy/_build_utils/_wrappers_common.py#L37-L40